tensorflow/core/kernels/mlir_generated/build_defs.bzl

```diff
@@ -284,16 +289,44 @@ def _gen_kernel_library(
       extra_args: Extra arguments to pass to the generator tool.
       test_tags: The tags to pass to the generated test.
       test_size: The "size" argument to pass to the test.
+      jit_i64_indexed_for_large_tensors_types: The input types for which to enable
+                                               JIT compilation of i64-indexed kernels for 
+                                               large inputs.
+      output_jit_i64_indexed_for_large_tensors_types: The output types for which to enable JIT 
+                                                      compilation of i64-indexed kernels for 
+                                                      large inputs.  
     """
 
     enable_cpu = bool(platform == "cpu")
-
-    aot_kernels = zip(types, output_types or types, [False for t in types])
-    jit_kernels = zip(jit_types, output_jit_types or jit_types, [True for t in jit_types])
-    all_kernels = aot_kernels + jit_kernels
-
+    if not output_types:
+        output_types = []
+    if not jit_types:
+        jit_types = []
+    if not output_jit_types:
+        output_jit_types = jit_types
+    if not jit_i64_indexed_for_large_tensors_types:
+        jit_i64_indexed_for_large_tensors_types = []
+    if not output_jit_i64_indexed_for_large_tensors_types:
+        output_jit_i64_indexed_for_large_tensors_types = jit_i64_indexed_for_large_tensors_types
+    # Fully JIT-compiled kernels
+    true_jits = [True for i in jit_types]
+    false_i64jits = [False for i in jit_types]
+    all_jit_kernels = zip(jit_types, output_jit_types,
+                          true_jits, false_i64jits)
+    # Partially JIT-compiled kernels
+    true_i64jits = [True for i in jit_i64_indexed_for_large_tensors_types]
+    false_jits = [True for i in jit_i64_indexed_for_large_tensors_types]
+    all_paratial_jit_kernels = zip(jit_i64_indexed_for_large_tensors_types,
+                                    output_jit_i64_indexed_for_large_tensors_types, 
+                                    false_jits, true_i64jits)
+    # AOT kernels
+    false_jits = [False for i in types]
+    aot_kernels = zip(types, output_types, false_jits, false_jits)
+    all_kernels = aot_kernels
+    all_kernels += all_jit_kernels
+    all_kernels += all_paratial_jit_kernels
```

Comment:
nit: This could probably go in one line: all_kernels = aot_kernels + all_jit_kernels + all_paratial_jit_kernels

Reply:
changed.

Reply:
Thank you Frederik, that actually fixed the ambiguous error message. However, seems like we have to decide between `gpu_atanh_kernels` and `gpu_atanh_kernels_experimental`. Bazel doesn't like them co-exist!

```Error in _gen_mlir_op_rule: _gen_mlir_op_rule rule 'generate_atanh_gpu_f32_f32_mlir' in package 'tensorflow/core/kernels/mlir_generated' conflicts with existing _gen_mlir_op_rule rule, defined at /workspace/sw3/LLVM/tensorflow/tensorflow-source/tensorflow/core/kernels/mlir_generated/BUILD:791:19
```
However, we have conditioned the choice based on `if_mlir_generated_experimental_kernels_enabled`. What do you think?

Reply:
Changed the naming contract, so hopefully it doesn't have any side effects.